### ISSUE TITLE
Incorrect determination that private is useless

### ISSUE THREAD

**Issue by ClayShentrup** [bbatsov/rubocop#3422]
Here's a simple example.

```ruby
module Foo
  extend ActiveSupport::Concern

  class_methods do
    private
    def some_private_class_method
      # ..
    end
  end

  private # Rubocop complains that this one is useless

  def some_private_instance_method
  end
end
```

## Expected behavior

The second `private` is fine because it's for instance methods, whereas the first `private` is for class methods.

## Actual behavior

The second `private` is determined to be useless.

## Steps to reproduce the problem

Run Rubocop on the example Ruby I just posted above.

## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.2.2 x86_64-darwin15)
```

**Comment by maxjacobson** [comment]
Hey @ClayShentrup,

I had a similar issue and made [a change to this cop to support your use-case](https://github.com/bbatsov/rubocop/pull/3186), which shipped in 0.42.0

Try adding this config to your .rubocop.yml:

```yaml
Lint/UselessAccessModifier:
  ContextCreatingMethods:
    - class_methods
```

Does that help?

**Comment by jonas054** [comment]
@maxjacobson's solution takes care of this particular problem, so I'm closing. Further enhancement of the `Lint/UselessAccessModifier` cop is discussed in #3483.
